src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -8632,6 +8632,20 @@ public override BoundNode VisitEventAssignmentOperator(BoundEventAssignmentOpera
             }
             VisitRvalue(node.Argument);
             // https://github.com/dotnet/roslyn/issues/31018: Check for delegate mismatch.
+            if (node.Argument.ConstantValue?.IsNull != true)
+            {
+                if (node.IsAddition)
+                {
+                    if (ResultType.IsNotNull)
+                    {
+                        LearnFromNonNullTest(MakeMemberSlot(receiverOpt, @event), ref State);
```

Comment:
>MakeMemberSlot(receiverOpt, @event) [](start = 45, length = 35)

Does `LearnFromNonNullTest()` handle an invalid slot? For instance, `F().E += () => { };`

Reply:
It seems like no, not really, and we don't have any debug asserts preventing e.g. `this.State[-1] = ...`. When we write it through to the BitVector, it ends up getting bit-shifted into something arbitrary and undesirable.

Reply:
Logged https://github.com/dotnet/roslyn/issues/46621 to fix `BitVector.this[int]`.

Reply:
have made the change to only try and update state if we have a valid member slot.